### ISSUE TITLE
Question: Add support info for using Parquet as storage

### ISSUE THREAD

**Issue by aaronsteers** [jwills/target-duckdb#20]
**Your question**

We have some users interested in storing data within Parquet. Can this target be used in combination with DuckDB's support for Parquet datasets?

**Comment by jwills** [comment]
Hey AJ-- I think @matsonj just used `target-parquet` for that in his MDS in a box project b/c of the (current) instability of the DuckDB file format, and then used the support for [external sources](https://github.com/jwills/dbt-duckdb#external-sources) in dbt-duckdb to do transformations on the resulting data.

I was just going to take a pass over this repo to do some updates for DuckDB 0.7.x-- is there some reason `target-parquet` wouldn't work for your user, or something that I could improve on it using DuckDB?

**Comment by aaronsteers** [comment]
Hi, @jwills. Re:

> is there some reason target-parquet wouldn't work for your user, or something that I could improve on it using DuckDB?

No reason I know of. As we think towards where to invest future efforts, and where to direct community members who want to interop with Spark and/or data lakes, I think DuckDB could ultimately be a better layer for "table-like" operations like merge/update/replace - since the SQL interop layer, in thoery, could better handle those more complex operations.

**Comment by jwills** [comment]
Yeah, your reasoning there re: upsert operations makes sense and is valid IMO. I'm going to turn my attention back to this project next week once I get some `dbt-duckdb` stuff I've been working on out the door and I will look hard at making parquet support for this target into a first-class concept.

**Comment by aaronsteers** [comment]
> Yeah, your reasoning there re: upsert operations makes sense and is valid IMO. 

Thanks for this validation! 

> I'm going to turn my attention back to this project next week once I get some `dbt-duckdb` stuff I've been working on out the door and I will look hard at making parquet support for this target into a first-class concept.

Sounds great. Again, no rush from our side. Nothing per se is broken as of now, and this is more of a long-term strategic investment, I think. 

I'll close this issue since the question is answered. Thanks again, and let us know if we can help in any way.

**Comment by jwills** [comment]
okay, cool-- as you can tell I've done ~ nothing to move this forward; do you want to chat about it somewhere? Meltano Slack?

**Comment by aaronsteers** [comment]
@jwills - great idea! https://meltano.slack.com/archives/C0514UNHQAX/p1680217170538049

(Join link for anyone not already in our slack: meltano.com/slack)

**Comment by aaronsteers** [comment]
Looks like @kgpayne has an implementation POC using external storage here:

- https://github.com/kgpayne/target-duckpond/pull/1

**Comment by ReneTC** [comment]
Any updates on this?
Not I'm running into the versioning error, with target-duckdb being older than the dbt-duckdb version.
Can the merged feature solve the issue? I'm not sure how to use it, I was looking for some documentation withuout luck.

**Comment by jwills** [comment]
@ReneTC I think the move is to use a virtualenv-type solution to align your duckdb, dbt-duckdb, and target-duckdb versions together; I'd recommend:

```
duckdb==0.8.1
dbt-duckdb==1.5.2
target-duckdb==0.6.0
```

...but I'm on vacation for a couple of weeks and haven't tried them in combination yet.

**Comment by ReneTC** [comment]
Thanks I'll test this tomorrow and report back
